src/Tasks/TaskTranslatorHelpers.cs

```diff
@@ -34,5 +35,94 @@ public static void Translate(this ITranslator translator, ref FrameworkName fram
+            {
+                dict = new Dictionary<string, DateTime>(comparer);
+                translator.Translate(ref count);
+                string key = string.Empty;
+                DateTime val = DateTime.Now;
+                for (int i = 0; i < count; i++)
+                {
+                    translator.Translate(ref key);
+                    translator.Translate(ref val);
+                    dict.Add(key, val);
+                }
+            }
+            else
+            {
+                count = dict.Count;
+                translator.Translate(ref count);
+                foreach (KeyValuePair<string, DateTime> kvp in dict)
+                {
+                    string key = kvp.Key;
+                    DateTime val = kvp.Value;
+                    translator.Translate(ref key);
+                    translator.Translate(ref val);
+                }
+            }
+        }
+
+        public static void Translate(this ITranslator translator, ref Dependencies dependencies, Type t)
+        {
+            translator.TranslateDictionary(ref dependencies.dependencies, (ITranslator translator, ref DependencyFile dependency) => {
+                if (t == typeof(ResGenDependencies.ResXFile))
+                {
+                    ResGenDependencies.ResXFile resx = dependency as ResGenDependencies.ResXFile;
+                    resx ??= new();
+                    translator.Translate(ref resx.linkedFiles);
+                    dependency = resx;
+                }
+                else if (t == typeof(ResGenDependencies.PortableLibraryFile))
+                {
+                    ResGenDependencies.PortableLibraryFile lib = dependency as ResGenDependencies.PortableLibraryFile;
+                    lib ??= new();
+                    translator.Translate(ref lib.assemblySimpleName);
+                    translator.Translate(ref lib.outputFiles);
+                    translator.Translate(ref lib.neutralResourceLanguage);
+                    dependency = lib;
+                }
+
+                dependency ??= new();
+                translator.Translate(ref dependency.filename);
+                translator.Translate(ref dependency.lastModified);
+                translator.Translate(ref dependency.exists);
+            });
+        }
+
+        public static void Translate(this ITranslator translator, ref StateFileBase stateFile, Type t)
+        {
+            if (t == typeof(ResGenDependencies))
+            {
+                ResGenDependencies rgd = stateFile as ResGenDependencies;
+                rgd ??= new();
+                translator.Translate(ref rgd.resXFiles, typeof(ResGenDependencies.ResXFile));
+                translator.Translate(ref rgd.portableLibraries, typeof(ResGenDependencies.PortableLibraryFile));
+                translator.Translate(ref rgd.baseLinkedFileDirectory);
+                stateFile = rgd;
+            }
+#if NETFRAMEWORK
+            else if (t == typeof(ResolveComReferenceCache))
```

Comment:
When called in dotnet core it will write nothing into file. If such file will be then deserialized in NETFRAMEWORK it will crash.
How do we protect against cache mangling between CLI and VS msbuild invocations?
I have not looked at caller side to verify that for given types it is never called for non NETFRAMEWORK, but even if it is assured, it makes code somehow fragile.

Reply:
It should only be .NET Framework --> .NET Framework. Those caches are only compiled, even, if the TFI is .NETFramework:
https://github.com/dotnet/msbuild/blob/main/src/Tasks/Microsoft.Build.Tasks.csproj#L558

If, at some point, we were to add support for those on .NET Core, we'd have to start by moving that to a not-framework-specific itemgroup, which would make the tests I wrote (and didn't specify not-core on) fail.